packages/terra-pills/src/FilterPills.jsx

```diff
@@ -0,0 +1,448 @@
+   * the screen and required for minimum accessibility standards. Providing this prop adds the `aria-label`
+   * attribute to the Pill List container element. (Required)
+   */
+  ariaLabel: PropTypes.string.isRequired,
+  /**
+   * If a visible text label is used with the collection of pills, provide a string of the ID for the html
+   * element containing the visible text label. Supplying the 'ariaLabelledBy' prop will override the 'ariaLabel'
+   * prop and adds the `aria-labelledby` attribute instead to the Pill List container element.
+   *
+   * ![IMPORTANT](https://badgen.net/badge/UX/Accessibility/blue) For best practices, ensure the visible text
+   * in the html element `id` provided to the Pill List `ariaLabelledby` prop matches the text provided to the
+   * `ariaLabel` prop, for consistency in the case of fallback or errors.
+   */
+  ariaLabelledBy: PropTypes.string,
+  /**
+   * If additional visible information text is used, provide a string containing the IDs for html elements that
+   * help describe the intent of the group of pills.
+   */
+  ariaDescribedBy: PropTypes.string,
+  /**
+   * The content to be shown in the pills container. Should only contain the sub-component `<Pills.Pill>`.
+   */
+  children: PropTypes.node,
+  /**
+   * Indicates if the Pills container is rolled up or not.
+   */
+  isCollapsible: PropTypes.bool,
+  /**
+   * Callback function to remove a pill, returns 'pillKey' and 'metadata' props from the removed `<Pills.Pill>` sub-component.
+   */
+  onRemove: PropTypes.func,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: PropTypes.shape({ formatMessage: PropTypes.func }).isRequired,
+};
+
+const defaultProps = {
+  ariaLabelledBy: undefined,
+  ariaDescribedBy: undefined,
+  children: undefined,
+  isCollapsible: false,
+};
+
+const FilterPills = (props) => {
+  const {
+    ariaLabel,
+    ariaLabelledBy,
+    ariaDescribedBy,
+    children,
+    intl,
+    isCollapsible,
+    onRemove,
+    ...customProps
+  } = props;
+
+  const theme = React.useContext(ThemeContext);
+  const [containerTabindex, setContainerTabindex] = useState('-1');
+  const [updatedCount, setUpdatedCount] = useState(React.Children.count(children));
+  const [rollUpCount, setRollUpCount] = useState(React.Children.count(children));
+  const [isCollapsed, setIsCollapsed] = useState(isCollapsible);
+  const [pillRemoved, setIsPillRemoved] = useState(false);
+  const [showRollupPillInteraction, setShowRollupPillInteraction] = useState(false);
+  const currentPill = useRef(); // ID of the pill that will recevie focus
+  const filterPillsRef = useRef();
+  const focusNode = useRef(0); // index of the pill that will receive focus
+  const isPillDeleted = useRef(false);
+  const isRollUpRemoved = useRef(false);
+  const removedLabel = useRef();
+
+  // Identifies the number of pills that needs to be hidden/rolled up
```

Comment:
This comment doesn't match the code. It looks like the number of pills to hide is determined in the utility. This is just setting it to collapsed or expanded.

Reply:
changed cfb6aad5b38a9957f1449aa2f75bbf5d001c3c96